### PR TITLE
feat(mdx): add tests for MDX editor and renderer components

### PR DESCRIPTION
**Pull request by dreyfus92**
#### Description

- Closes # 
- What does this PR change? Give us a brief description.

## Summary by CodeRabbit

- Tests
  - Added comprehensive automated tests for the MDX editor, renderer, plugin setup, and render pipeline covering typical and edge cases.
  - Added test utilities and a package-level Vitest configuration to enable consistent MDX package testing.

- Chores
  - Added a test script to the MDX package and included the MDX package in workspace test runs.

- Public API
  - Added a new exported integration helper to simplify MDX integration configuration.

### CODE REVIEW

File: packages/@studiocms/mdx/vitest.config.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export default defineProject(
+	getViteConfig(
+		{
+			test: {
+				environment: 'node',
+				include: ['**/*.test.ts'],
+			},
+		},
+		{
+			integrations: [testIntegration, studiocmsMDX()],
```
Comment (Adammatthiesen) [inline packages/@studiocms/mdx/vitest.config.ts]:
```suggestion
			integrations: [testIntegration],
```

You can't use a StudioCMS Plugin as an Astro integration, You'll need to extract the integration like how the tests for MD/HTML do.

**Review by Adammatthiesen — CHANGES_REQUESTED** [review]
